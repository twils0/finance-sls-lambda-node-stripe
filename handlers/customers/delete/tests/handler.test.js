const { loadStripe } = require('../../../../functions/stripe/loadStripe');
const { deleteCustomer } = require('../../../../functions/stripe/deleteCustomer');

const deleteExport = require('../handler');

const deleteModule = deleteExport.delete;

global.console.log = jest.fn();
jest.mock('../../../../functions/stripe/loadStripe', () => ({
  loadStripe: jest.fn(),
}));
jest.mock('../../../../functions/stripe/deleteCustomer', () => ({
  deleteCustomer: jest.fn(),
}));
jest.mock('../../../../functions/errorResponse', () => ({
  errorResponse: jest.fn(error => error),
}));

const event = {
  cognitoPoolClaims: {
    stripe_customer_id: 'testCustomerID',
  },
};
const response = {
  status: 200,
  body: {
    message: 'success',
  },
};
const callback = jest.fn();

describe('handlers', () => {
  describe('users', () => {
    describe('delete', () => {
      afterEach(() => {
        loadStripe.mockReset();
        deleteCustomer.mockReset();
        global.console.log.mockReset();
        callback.mockReset();
      });

      it('fails and returns error, when loadStripe throws error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        loadStripe.mockReturnValue(Promise.reject(error));

        await deleteModule(event, null, callback);

        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('fails and returns error, when deleteCustomer throws error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        loadStripe.mockReturnValue(Promise.resolve());
        deleteCustomer.mockReturnValue(Promise.reject(error));

        await deleteModule(event, null, callback);

        expect(deleteCustomer).toBeCalledWith(event.cognitoPoolClaims.stripe_customer_id);
        expect(callback).toBeCalledWith(error, response);
        expect(global.console.log).toMatchSnapshot();
      });

      it('correctly deletes user', async () => {
        loadStripe.mockReturnValue(Promise.resolve());
        deleteCustomer.mockReturnValue(Promise.resolve());

        await deleteModule(event, null, callback);

        expect(deleteCustomer).toBeCalledWith(event.cognitoPoolClaims.stripe_customer_id);
        expect(callback).toBeCalledWith(null, response);
        expect(global.console.log).toMatchSnapshot();
      });
    });
  });
});
