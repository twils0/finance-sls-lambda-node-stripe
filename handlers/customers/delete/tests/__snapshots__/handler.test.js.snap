// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handlers users delete correctly deletes user 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "loadStripe
",
    ],
    Array [
      "deleteCustomer
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users delete fails and returns error, when deleteCustomer throws error 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "loadStripe
",
    ],
    Array [
      "deleteCustomer
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`handlers users delete fails and returns error, when loadStripe throws error 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "loadStripe
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
